app/routes.py

```diff
@@ -1,2 +1,272 @@
+def valid_int(number,parameter_type):
+    try:
+        int(number)
+    except:
+        abort(make_response({"error":f"{parameter_type} must be an int"},400))
+
+@tasks_bp.route("", methods=["GET", "POST"])
+def handle_tasks():
+# Wave 1: Get Tasks: Getting Saved Tasks
+    if request.method == "GET":
+        sort = request.args.get("sort")
+        if sort == "asc":
+            tasks = Task.query.order_by(Task.title)
+        elif sort == "desc":
+            tasks = Task.query.order_by(Task.title.desc())
+        else:
+            tasks = Task.query.all()
+#Wave 1: Get Tasks: No Saved Tasks
+        tasks_response = []
+        for task in tasks:
+            has_complete = task.completed_at
+            tasks_response.append(
+                {
+                    "description": task.description,
+                    "id": task.id,
+                    "is_complete": False if has_complete == None else has_complete,
+                    "title": task.title,
+                }
+            )
+        return jsonify(tasks_response)
+# Wave 1: Create a Task: Valid Task With null completed_at
+    elif request.method == "POST":
+        request_body = request.get_json()
+#Wave 1: Create A Task: Missing Title
+        if "title" not in request_body or "description" not in request_body or "completed_at" not in request_body:
+            return jsonify ({
+                "details": "Invalid data"
+            }), 400
+
+
+        new_task = Task(
+            title=request_body["title"],
+            description=request_body["description"],
+            completed_at=request_body["completed_at"]
+        )
+        db.session.add(new_task)
+        db.session.commit()
+        
+#Wave 1: Create A Task: Valid Task with null completed_at 201 CREATED
+
+
+        return jsonify({"task":new_task.to_dict()}),201
+# handle_one_task handles GET,PUT and DELETE requests for the tasks/task_id endpoint
+@tasks_bp.route("/<task_id>", methods=["GET", "PUT", "DELETE"])
+def handle_one_task(task_id):
+    valid_int(task_id,"task_id")
+    task = Task.query.get_or_404(task_id)
+# Wave 1: Get One Task: One Saved Task
+    if request.method == "GET":
+        has_complete = task.completed_at
+        if task.goal_id:
+            task_response={   
+                    "task": {
+                    "id": task.id,
+                    "goal_id": task.goal_id,
+                    "title": task.title,
+                    "description": task.description,
+                    "is_complete": False if has_complete == None else has_complete,
+                    
+                }
+                }
+        else:
+            task_response={   
+                    "task": {
+                    "id": task.id,
+                    "title": task.title,
+                    "description": task.description,
+                    "is_complete": False if has_complete == None else has_complete,
+                    
+                }
+                }
```

Comment:
Task to_dict handles the existence of goal_id.
```suggestion
        task_response = {"task":task.to_dict()}
```